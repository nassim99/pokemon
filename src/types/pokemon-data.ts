export type PokemonDataProps = {
  id: number;
  name: string;
  height: number;
  weight: number;
  sprites: {
    front_default: string;
  };
  types: Array<{
    type: {
      name: string;
    };
  }>;
  moves: Array<{
    move: {
      name: string;
    };
  }>;
  abilities: Array<{
    ability: {
      name: string;
    };
    is_hidden: boolean;
  }>;
  species: {
    gender_rate: number;
    capture_rate: number;
    flavor_text_entries: Array<{
      flavor_text: string;
    }>;
  };
};
