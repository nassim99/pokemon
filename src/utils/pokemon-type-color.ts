export const pokemonColor: (t: string) => string = type => {
  const bg =
    type in PokemonTypeColors
      ? (PokemonTypeColors[type as keyof typeof PokemonTypeColors] as string)
      : PokemonTypeColors['normal'];
  return bg;
};

export const PokemonTypeColors = {
  normal: '#CDCDB9',
  fire: '#F4934D',
  fighting: '#BA5852',
  water: '#85A5F0',
  flying: '#B8A5F2',
  grass: '#99D07D',
  poison: '#A768A7',
  electric: '#F9DF78',
  ground: '#EDD081',
  psychic: '#F47DA1',
  rock: '#C5B059',
  ice: '#B3E1E1',
  bug: '#B5C534',
  dragon: '#8656FA',
  ghost: '#7D6B9B',
  dark: '#756459',
  steel: '#C1C1D1',
  fairy: '#EFA7B7',
};
