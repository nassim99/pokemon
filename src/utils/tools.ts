export const numberToMeter: (n: number) => string = (n: number) => n / 10 + 'm';
export const numberToKg: (n: number) => string = (n: number) => n / 10 + 'Kg';
