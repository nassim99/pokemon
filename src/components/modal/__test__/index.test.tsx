import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Modal from '../index';
import data from './pokemon.json';
import { PokemonDataProps } from 'types/pokemon-data';
import { pokemonColor } from 'utils/pokemon-type-color';

let pokemon: undefined | PokemonDataProps = data;
const closeModal = jest.fn(() => (pokemon = undefined));

describe('Modal Component', () => {
  beforeEach(() => {
    render(<Modal pokemon={pokemon} onClose={closeModal} />);
  });

  it('should modal render pokemon data', () => {
    expect(screen.getByText('Description')).toBeInTheDocument();
    expect(screen.getByText('bulbasaur')).toBeInTheDocument();
    expect(screen.getByTestId('header')).toHaveStyle('background-color: ' + pokemonColor('grass'));
  });
  it('test close button', async () => {
    expect(pokemon).not.toBe(undefined);
    expect(screen.getByText('bulbasaur')).toBeInTheDocument();
    fireEvent.click(screen.getByTestId('closeButton'));
    expect(closeModal).toBeCalledTimes(1);
  });
  it('test matched snapshot', async () => {
    expect(screen.getByTestId('modalContainer')).toMatchSnapshot();
  });
});
