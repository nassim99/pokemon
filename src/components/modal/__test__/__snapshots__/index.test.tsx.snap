// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component test matched snapshot 1`] = `
<section
  class="modal-container"
  data-testid="modalContainer"
  id="modal"
>
  <div
    class="modal-content"
  >
    <header
      class="headerModal"
      data-testid="header"
      style="background-color: rgb(153, 208, 125);"
    >
      <span
        class="pokemonId"
      >
        N: #
        1
      </span>
      <h1>
        bulbasaur
      </h1>
      <img
        class="flex imageContainer"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
      />
      <div
        class="closeWrapper"
        data-testid="closeButton"
      >
        <svg
          stroke="#ebebeb"
          stroke-width="10"
        >
          close.svg
        </svg>
      </div>
    </header>
    <div
      class="paddingHorizontal"
    >
      <div>
        <h2
          class="paddingHorizontal"
          style="background-color: rgb(153, 208, 125);"
        >
          Description
        </h2>
        <p
          class="paddingHorizontal"
        >
          A strange seed was
planted on its
back at birth.The plant sprouts
and grows with
this POKéMON.
        </p>
      </div>
      <div>
        <h2
          class="paddingHorizontal"
          style="background-color: rgb(153, 208, 125);"
        >
          Informations
        </h2>
        <div
          class="paddingHorizontal flex"
        >
          <div
            class="flex05"
          >
            <div>
              <b>
                Type: 
              </b>
              <span
                class="typeName"
                style="color: rgb(153, 208, 125);"
              >
                grass
              </span>
              <span
                class="typeName"
                style="color: rgb(167, 104, 167);"
              >
                poison
              </span>
            </div>
            <div>
              <b>
                HEIGHT:
              </b>
               
              0.7m
            </div>
            <div>
              <b>
                WEIGHT:
              </b>
               
              6.9Kg
            </div>
          </div>
          <div
            class="flex05"
          >
            <div>
              <b>
                CATCH RATE:
              </b>
               
              45
              %
            </div>
            <div>
              <b>
                GENDER:
              </b>
               ♀ 
              12.5
              % | ♂ 
              87.5
              %
            </div>
            <div>
              <b>
                ABILITIES: 
              </b>
              <span>
                overgrow,  
              </span>
              <span>
                chlorophyll (ability hidden) 
              </span>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h2
          class="paddingHorizontal"
          style="background-color: rgb(153, 208, 125);"
        >
          Moves
        </h2>
        <div
          class="flex movesWrapper paddingHorizontal"
          data-testid="moves"
        >
          <div
            class="moveName"
          >
            razor-wind
          </div>
          <div
            class="moveName"
          >
            swords-dance
          </div>
          <div
            class="moveName"
          >
            cut
          </div>
          <div
            class="moveName"
          >
            bind
          </div>
          <div
            class="moveName"
          >
            vine-whip
          </div>
          <div
            class="moveName"
          >
            headbutt
          </div>
          <div
            class="moveName"
          >
            tackle
          </div>
          <div
            class="moveName"
          >
            body-slam
          </div>
          <div
            class="moveName"
          >
            take-down
          </div>
          <div
            class="moveName"
          >
            double-edge
          </div>
          <div
            class="moveName"
          >
            growl
          </div>
          <div
            class="moveName"
          >
            strength
          </div>
          <div
            class="moveName"
          >
            mega-drain
          </div>
          <div
            class="moveName"
          >
            leech-seed
          </div>
          <div
            class="moveName"
          >
            growth
          </div>
          <div
            class="moveName"
          >
            razor-leaf
          </div>
          <div
            class="moveName"
          >
            solar-beam
          </div>
          <div
            class="moveName"
          >
            poison-powder
          </div>
          <div
            class="moveName"
          >
            sleep-powder
          </div>
          <div
            class="moveName"
          >
            petal-dance
          </div>
          <div
            class="moveName"
          >
            string-shot
          </div>
          <div
            class="moveName"
          >
            toxic
          </div>
          <div
            class="moveName"
          >
            rage
          </div>
          <div
            class="moveName"
          >
            mimic
          </div>
          <div
            class="moveName"
          >
            double-team
          </div>
          <div
            class="moveName"
          >
            defense-curl
          </div>
          <div
            class="moveName"
          >
            light-screen
          </div>
          <div
            class="moveName"
          >
            reflect
          </div>
          <div
            class="moveName"
          >
            bide
          </div>
          <div
            class="moveName"
          >
            sludge
          </div>
          <div
            class="moveName"
          >
            skull-bash
          </div>
          <div
            class="moveName"
          >
            amnesia
          </div>
          <div
            class="moveName"
          >
            flash
          </div>
          <div
            class="moveName"
          >
            rest
          </div>
          <div
            class="moveName"
          >
            substitute
          </div>
          <div
            class="moveName"
          >
            snore
          </div>
          <div
            class="moveName"
          >
            curse
          </div>
          <div
            class="moveName"
          >
            protect
          </div>
          <div
            class="moveName"
          >
            sludge-bomb
          </div>
          <div
            class="moveName"
          >
            mud-slap
          </div>
          <div
            class="moveName"
          >
            giga-drain
          </div>
          <div
            class="moveName"
          >
            endure
          </div>
          <div
            class="moveName"
          >
            charm
          </div>
          <div
            class="moveName"
          >
            swagger
          </div>
          <div
            class="moveName"
          >
            fury-cutter
          </div>
          <div
            class="moveName"
          >
            attract
          </div>
          <div
            class="moveName"
          >
            sleep-talk
          </div>
          <div
            class="moveName"
          >
            return
          </div>
          <div
            class="moveName"
          >
            frustration
          </div>
          <div
            class="moveName"
          >
            safeguard
          </div>
          <div
            class="moveName"
          >
            sweet-scent
          </div>
          <div
            class="moveName"
          >
            synthesis
          </div>
          <div
            class="moveName"
          >
            hidden-power
          </div>
          <div
            class="moveName"
          >
            sunny-day
          </div>
          <div
            class="moveName"
          >
            rock-smash
          </div>
          <div
            class="moveName"
          >
            facade
          </div>
          <div
            class="moveName"
          >
            nature-power
          </div>
          <div
            class="moveName"
          >
            ingrain
          </div>
          <div
            class="moveName"
          >
            knock-off
          </div>
          <div
            class="moveName"
          >
            secret-power
          </div>
          <div
            class="moveName"
          >
            grass-whistle
          </div>
          <div
            class="moveName"
          >
            bullet-seed
          </div>
          <div
            class="moveName"
          >
            magical-leaf
          </div>
          <div
            class="moveName"
          >
            natural-gift
          </div>
          <div
            class="moveName"
          >
            worry-seed
          </div>
          <div
            class="moveName"
          >
            seed-bomb
          </div>
          <div
            class="moveName"
          >
            energy-ball
          </div>
          <div
            class="moveName"
          >
            leaf-storm
          </div>
          <div
            class="moveName"
          >
            power-whip
          </div>
          <div
            class="moveName"
          >
            captivate
          </div>
          <div
            class="moveName"
          >
            grass-knot
          </div>
          <div
            class="moveName"
          >
            venoshock
          </div>
          <div
            class="moveName"
          >
            round
          </div>
          <div
            class="moveName"
          >
            echoed-voice
          </div>
          <div
            class="moveName"
          >
            grass-pledge
          </div>
          <div
            class="moveName"
          >
            work-up
          </div>
          <div
            class="moveName"
          >
            grassy-terrain
          </div>
          <div
            class="moveName"
          >
            confide
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
